build/Makefile

```diff
@@ -443,42 +446,41 @@ setup-prometheus: PVC ?= true
 setup-prometheus: PV_SIZE ?= 64Gi
 setup-prometheus: SCRAPE_INTERVAL=30s
 setup-prometheus:
-	$(DOCKER_RUN) \
-		helm upgrade --install --wait prom stable/prometheus --namespace metrics \
+	$(DOCKER_RUN) helm repo add stable https://kubernetes-charts.storage.googleapis.com/
+	$(DOCKER_RUN) helm repo update
+	$(DOCKER_RUN) helm upgrade prom stable/prometheus \
+		--install --wait --namespace metrics --create-namespace \
 		--set server.global.scrape_interval=$(SCRAPE_INTERVAL),server.persistentVolume.enabled=$(PVC),server.persistentVolume.size=$(PV_SIZE) \
 		$(HELM_ARGS) \
 		-f $(mount_path)/build/prometheus.yaml
 
 uninstall-prometheus: $(ensure-build-image)
 	$(DOCKER_RUN) \
-		helm delete --purge prom
+		helm uninstall prom --namespace=metrics
 
 # setup grafana in the current cluster with datasource and dashboards ready for use with agones
 # by default Persistent Volume Claims are requested.
 setup-grafana: PVC ?= true
 setup-grafana: PV_SIZE ?= 64Gi
 setup-grafana: PASSWORD ?= admin
 setup-grafana:
+	-$(DOCKER_RUN) kubectl create namespace metrics
+	$(DOCKER_RUN) helm repo add stable https://kubernetes-charts.storage.googleapis.com/
+	$(DOCKER_RUN) helm repo update
 	$(DOCKER_RUN) kubectl apply -f $(mount_path)/build/grafana/
-	$(DOCKER_RUN) \
-		helm upgrade --install --wait grafana stable/grafana --version=5.0.13  --namespace metrics \
+	$(DOCKER_RUN) helm upgrade grafana stable/grafana \
+		--install --wait --namespace metrics \
 		--set persistence.enabled=$(PVC),server.persistentVolume.size=$(PV_SIZE) \
 		--set adminPassword=$(PASSWORD) -f $(mount_path)/build/grafana.yaml
 
 uninstall-grafana: $(ensure-build-image)
 	$(DOCKER_RUN) \
-		helm delete --purge grafana
+		helm uninstall grafana --namespace=metrics
```

Comment:
Should this remove the resources that were created by the kubectl apply command as well?

Reply:
Yeah, we should.

Reply:
Do you want to fix that in this PR or in a subsequent one?

(since this is an improvement separate from upgrading to helm 3 and not a regression and I don't want to slow you down too much)

Reply:
Already done it :smile: just needed a `$(DOCKER_RUN) kubectl delete -f $(mount_path)/build/grafana/` will push up in a minute.